### PR TITLE
Hide HEX freeclaim section if wallet has nothing to claim or has claimed

### PR DESCRIPTION
**Pull request by Mvii7**

### CODE REVIEW

**Comment by Mvii7** [comment]
This should ideally be tested on a staging environment before deploying to prod. We need a staging environment that has the snapshot API and is using the ETH mainnet

**Review by Palmerg4 — APPROVED** [review]
Looks like a nice fix

**Review by GeertWille — APPROVED** [review]

**Review by GeertWille — APPROVED** [review]

**Comment by Mvii7** [comment]
This one is a bit difficult to test, maybe we should only hide if already claimed, and ask whoever was eligible to claim, check if they can see the section. Otherwise we would need to find someone who hasn't claimed yet and ask if they can see the section. I'm not eligible so I won't see regardless.

**Comment by livinglikemike** [comment]
> 
> 
> This one is a bit difficult to test, maybe we should only hide if already claimed, and ask whoever was eligible to claim, check if they can see the section. Otherwise we would need to find someone who hasn't claimed yet and ask if they can see the section. I'm not eligible so I won't see regardless. Let me know your thoughts.

I have a wallet that has not claimed yet as a hex holder. I can test it with that one.

**Comment by Mvii7** [comment]
> > This one is a bit difficult to test, maybe we should only hide if already claimed, and ask whoever was eligible to claim, check if they can see the section. Otherwise we would need to find someone who hasn't claimed yet and ask if they can see the section. I'm not eligible so I won't see regardless. Let me know your thoughts.
> 
> I have a wallet that has not claimed yet as a hex holder. I can test it with that one.

Awesome, lets get this branch onto staging and test

**Review by Vicarious1 — APPROVED** [review]
Looks good to me!

File: src/app/claim-page/claim-page.component.html [hunk 1]
```diff
@@ -182,8 +182,7 @@
     </div>
   </div>
 </section>
-
-<section class="axn-section" *ngIf="account">
+<section *ngIf="account ? account.snapshot?.nothingToClaim === false || account.completeClaim?.hasClaimed === false : false" class="axn-section">
```
Comment (Mvii7) [inline src/app/claim-page/claim-page.component.html:185]: This logic is off, testing with mocked up data - fixing.

Comment (Mvii7) [reply src/app/claim-page/claim-page.component.html]: Fixed, just needed && instead of ||

**Comment by Mvii7** [comment]
I tested locally - works as intended

File: src/app/services/contract/index.ts [hunk 2]
```diff
@@ -1553,9 +1553,9 @@ export class ContractService {
         .then(
           (result) => {
             this.account.snapshot = result;
-            this.account.snapshot.user_dont_have_hex =
+            this.account.snapshot.nothingToClaim =
               this.account.snapshot.hex_amount <= 0;
```
Comment (Mvii7) [inline src/app/services/contract/index.ts:1557]: hex_amount comes from the Hex snapshot API

**Pull request merged**